src/config.js

```diff
@@ -0,0 +1,156 @@
+  _debug: DEFAULT_DEBUG,
+  get debug() {
+    if ($$PREBID_GLOBAL$$.logging || $$PREBID_GLOBAL$$.logging === false) {
+      return $$PREBID_GLOBAL$$.logging;
+    }
+    return this._debug;
+  },
+  set debug(val) {
+    this._debug = val;
+  },
+
+  // default timeout for all bids
+  _bidderTimeout: DEFAULT_BIDDER_TIMEOUT,
+  get bidderTimeout() {
+    return $$PREBID_GLOBAL$$.bidderTimeout || this._bidderTimeout;
+  },
+  set bidderTimeout(val) {
+    this._bidderTimeout = val;
+  },
+
+  // domain where prebid is running for cross domain iframe communication
+  _publisherDomain: DEFAULT_PUBLISHER_DOMAIN,
+  get publisherDomain() {
+    return $$PREBID_GLOBAL$$.publisherDomain || this._publisherDomain;
+  },
+  set publisherDomain(val) {
+    this._publisherDomain = val;
+  },
+
+  // delay to request cookie sync to stay out of critical path
+  _cookieSyncDelay: DEFAULT_COOKIESYNC_DELAY,
+  get cookieSyncDelay() {
+    return $$PREBID_GLOBAL$$.cookieSyncDelay || this._cookieSyncDelay;
+  },
+  set cookieSyncDelay(val) {
+    this._cookieSyncDelay = val;
+  },
+
+  // calls existing function which may be moved after deprecation
+  set priceGranularity(val) {
+    $$PREBID_GLOBAL$$.setPriceGranularity(val);
+  },
+
+  _sendAllBids: DEFAULT_ENABLE_SEND_ALL_BIDS,
+  get enableSendAllBids() {
+    return this._sendAllBids;
+  },
+  set enableSendAllBids(val) {
+    this._sendAllBids = val;
+  },
+
+  // calls existing function which may be moved after deprecation
+  set bidderSequence(val) {
+    $$PREBID_GLOBAL$$.setBidderSequence(val);
+  },
+
+  // calls existing function which may be moved after deprecation
+  set s2sConfig(val) {
+    $$PREBID_GLOBAL$$.setS2SConfig(val);
+  },
+};
+
+/*
+ * Returns configuration object or single configuration property if given
+ * a string matching a configuartion property name
+ */
+export function getConfig(option) {
+  return option ? config[option] : config;
+}
+
+/*
+ * Sets configuration given an object containing key-value pairs and calls
+ * listeners that were added by the `subscribe` function
+ */
+export function setConfig(options) {
+  if (typeof options !== 'object') {
+    utils.logError('setConfig options must be an object');
+  }
+
+  Object.assign(config, options);
+  callSubscribers(options);
+}
+
+/*
+ * Adds a function to a set of listeners that are invoked whenever `setConfig`
+ * is called. The subscribed function will be passed the options object that
+ * was used in the `setConfig` call. Topics can be subscribed to to only get
+ * updates when specific properties are updated by passing a topic string as
+ * the first parameter.
+ *
+ * Example use:
+ * // subscribe to all configuration changes
+ * subscribe((config) => console.log('config set:', config));
+ *
+ * // subscribe to only 'logging' changes
+ * subscribe('logging', (config) => console.log('logging set:', config));
+ */
+export function subscribe(topic, listener) {
```

Comment:
I like how this `subscribe` is implemented now.  👍 

However, I think it might be nice to merge its functionality with `getConfig` for a few reasons.  For one, I think that `getConfig` is a more descriptive name, especially if we export this on the public API (which I think we should).  Also, it really is fulfilling the same purpose as `getConfig`; it's just implemented as a push rather than a pull.

I propose we merge `subscribe` into `getConfig`.  if `getConfig` is called with no arguments or string, behave as `getConfig` does now, if it's called with a function or a string and a function, then it behaves as `subscribe` behaves now.